### PR TITLE
RC V2.7.0

### PR DESCRIPTION
**Pull request by aburrell**
# Description

This is a release that updates the coefficients for IGRF-14 and also contains the new version of the C code, 2.7.  Deprecated functions have been removed.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality
      to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Ran unit tests locally after updating them for new behaviour.

**Test Configuration**:
* Operating system: OS X Ventura
* Version number: Python 3.10
* Any details about your local setup that are relevant: gcc 14.2.0

# Checklist:

- [x] Update the version number
- [x] Verify and update Zenodo information in .zenodo.json
- [x] doc tests pass locally
- [x] activate this branch on readthedocs (see CI tests below)
- [x] Upload the test distribution to the testPyPi server: https://test.pypi.org/project/aacgmv2/2.7.0rc1/
- [x] Contacted developers and users about the future release
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``Changelog.rst``, summarising the changes
- [x] Add yourself to ``AUTHORS.rst`` and ``.zenodo.json``

### CODE REVIEW

**Comment by aburrell** [comment]
@dchaddock, @MITHaystack, @JouleCai, @indiajacksonphd, @sapols, @KCollins, @dchaddock, @mattkjames7, @mshumko, @HamSCI, @jklenzing, @aalekseevx, and @zemarchezi.  Each of you have actively developed projects that depend on the most recent version of AACGMV2.  If you would like, here's the release candidate you can test out.  If you have any issues, feel free to let me know. 

I am going to try and get this out before the new year to prevent any issues with realtime processing in AAGCMV2 coordinates due to the IGRF-14 update.

**Comment by jklenzing** [comment]
Updated my RC tests in github actions [here](https://github.com/pysat/pysatMissions/pull/124). Tests are successfully running with the RC as a dependency, however I am getting the following warning from aacgmv2: `DeprecationWarning: path is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.` Not sure if this is expected in this release.

File: aacgmv2/__init__.py [hunk 1]
```diff
@@ -47,10 +48,14 @@
...
-IGRF_COEFFS = _os.path.join(_os.path.realpath(_os.path.dirname(__file__)),
-                            'magmodel_1590-2020.txt')
+AACGM_v2_DAT_PREFIX = _os.path.join(
+    _os.path.split(str(resources.path(__package__,
+                                      '__init__.py').__enter__()))[0],
+    'aacgm_coeffs', 'aacgm_coeffs-14-')
+IGRF_COEFFS = _os.path.join(
+    _os.path.split(str(resources.path(__package__,
+                                      '__init__.py').__enter__()))[0],
+    'magmodel_1590-2025.txt')
```
Comment (jklenzing) [inline aacgmv2/__init__.py:58]: These are the lines that generate the Deprecation Warning. `resources.path` is replaced by `resources.files`. Probably worth an issue for the next update.

File: setup.cfg [hunk 2]
```diff
@@ -1,96 +1,8 @@
...
-
-[options]
-zip_safe = False
-install_requires = numpy
-include_package_data = True
-include_entry_points = True
-
-[aliases]
-release = register clean --all sdist
+version = 2.7.0
```
Comment (jklenzing) [inline setup.cfg:3]: I think you could drop the duplicate metadata here since python 3.9 is the minimum supported version.

**Review by jklenzing — COMMENTED** [review]
Overall looks good. A few minor comments on things that could just be written up as issues for a future release.

File: setup.cfg (hunk 2 above)
Comment (aburrell) [reply setup.cfg:3]: #95

File: aacgmv2/__init__.py (hunk 1 above)
Comment (aburrell) [reply aacgmv2/__init__.py:58]: #94

**Review by jklenzing — APPROVED** [review]

**Pull request merged**